services/QuillLMS/engines/evidence/lib/evidence/evidence/prefilter_check.rb

```diff
@@ -0,0 +1,53 @@
+module Evidence
+  class PrefilterCheck
+    attr_accessor :prefilter_rules
+    MINIMUM_WORD_COUNT = 3
+    OPTIMAL_RULE_UID = '' # TODO: populate
+
+    PREFILTERS = {
+      # TODO: populate
+    }
+
+    def initialize(entry)
+      @entry = entry
+      @prefilter_rules = Evidence::Rule.where(rule_type: Evidence::Rule::TYPE_PREFILTER).includes(:feedbacks)
+      
+    end
+
+    def default_response
+      {
+        feedback: '',
+        feedback_type: Evidence::Rule::TYPE_PREFILTER,
+        optimal: true,
+        response_id: '',
+        entry: @entry,
+        concept_uid: '',
+        rule_uid: OPTIMAL_RULE_UID,
+        highlight: []
+      }
+    end
+
+    def feedback_object
+      violated_rule = @prefilter_rules.find do |rule| 
+        next unless PREFILTERS[rule.uid]
+        !PREFILTERS[rule.uid].call(@entry) 
+      end
+      return default_response unless violated_rule
+
+      feedback = violated_rule.feedbacks.first
```

Comment:
Not for this PR, but I wonder if in the long term we actually want some sort of logic to live on the `Rule` model that returns the appropriate `Feedback` object.  I'm assuming that such logic is pretty universal and not tied to specific algorithms.  I could imagine something like `Rule.get_feedback()`.